app/scripts/views/mixins/password-mixin.js

```diff
@@ -10,7 +10,8 @@ define(function (require, exports, module) {
 
   module.exports = {
     events: {
-      'change .show-password': 'onPasswordVisibilityChange'
+      'change .show-password': 'onPasswordVisibilityChange',
+      'keypress input.password': 'onPasswordKeypress'
```

Comment:
I think doing `keyup` instead of `keypress` might make the code a little easier to follow. Getting the input length after this should give you a better indication of how long the password being entered in really is.